tests/unit/amo/components/TestAddonRecommendations.js

```diff
@@ -0,0 +1,224 @@
+import * as React from 'react';
+
+import AddonRecommendations, {
+  AddonRecommendationsBase,
+  TAAR_IMPRESSION_CATEGORY,
+  TAAR_COHORT_COOKIE_NAME,
+  TAAR_COHORT_INCLUDED,
+  TAAR_COHORT_EXCLUDED,
+} from 'amo/components/AddonRecommendations';
+import {
+  fetchRecommendations,
+  loadRecommendations,
+} from 'amo/reducers/recommendations';
+import { createInternalAddon } from 'core/reducers/addons';
+import {
+  dispatchClientMetadata,
+  fakeAddon,
+  fakeRecommendations,
+} from 'tests/unit/amo/helpers';
+import {
+  createStubErrorHandler,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+const fakeCookie = (returnValue) => {
+  return {
+    load: sinon.stub().returns(returnValue),
+    save: sinon.spy(),
+  };
+};
+
+const fakeRandomizer = (expectedRecommendedValue) => {
+  return sinon.stub().returns(expectedRecommendedValue ? 0.51 : 0.5);
+};
+
+describe(__filename, () => {
+  let fakeTracking;
+  let store;
+
+  beforeEach(() => {
+    fakeTracking = {
+      sendEvent: sinon.spy(),
+    };
+    store = dispatchClientMetadata().store;
+  });
+
+  function doFetchRecommendations({
+    guid = fakeAddon.guid,
+    recommended = true,
+  }) {
+    const errorHandler = createStubErrorHandler();
+    return fetchRecommendations({
+      errorHandlerId: errorHandler.id,
+      guid,
+      recommended,
+    });
+  }
+
+  function doLoadRecommendations({
+    addon = createInternalAddon(fakeAddon),
+    ...props
+  }) {
+    return loadRecommendations({
+      guid: addon.guid,
+      ...fakeRecommendations,
+      ...props,
+    });
+  }
+
+  function render({
+    addon = createInternalAddon(fakeAddon),
+    cookie = fakeCookie(),
+    randomizer = fakeRandomizer(true),
+    ...props
+  } = {}) {
+    const errorHandler = createStubErrorHandler();
+
+    return shallowUntilTarget(
+      <AddonRecommendations
+        addon={addon}
+        i18n={fakeI18n()}
+        cookie={cookie}
+        errorHandler={errorHandler}
+        randomizer={randomizer}
+        store={store}
+        tracking={fakeTracking}
+        {...props}
+      />,
+      AddonRecommendationsBase
+    );
+  }
+
+  it('renders nothing without an addon', () => {
+    const root = render({ addon: null });
+    expect(root).not.toHaveClassName('AddonRecommendations');
+  });
+
+  it('renders an AddonCard when recommendations are loaded', () => {
+    const addons = [createInternalAddon(fakeAddon)];
```

Comment:
I thought we received `Array<ExternalAddonType>` in `loadRecommendations()`?

Reply:
Good catch, thanks!